tests/tenant-fetcher/tests/handler_test.go

```diff
@@ -163,27 +189,221 @@ func TestDecommissioningHandler(t *testing.T) {
+		}
+
+		addRegionalTenantExpectStatusCode(t, providedTenant, http.StatusOK)
+
+		oldTenantState, err := fixtures.GetTenants(dexGraphQLClient)
+		require.NoError(t, err)
+
+		removeTenantExpectStatusCode(t, providedTenant, http.StatusOK)
+
+		newTenantState, err := fixtures.GetTenants(dexGraphQLClient)
+		require.NoError(t, err)
+
+		// THEN
+		assert.Equal(t, len(oldTenantState), len(newTenantState))
+	})
+}
+
 func addTenantExpectStatusCode(t *testing.T, providedTenant Tenant, expectedStatusCode int) {
-	makeTenantRequestExpectStatusCode(t, providedTenant, http.MethodPut, expectedStatusCode)
+	makeTenantRequestExpectStatusCode(t, providedTenant, http.MethodPut, config.TenantFetcherFullURL, expectedStatusCode)
+}
+
+func addRegionalTenantExpectStatusCode(t *testing.T, providedTenant Tenant, expectedStatusCode int) {
+	makeTenantRequestExpectStatusCode(t, providedTenant, http.MethodPut, config.TenantFetcherFullRegionalURL, expectedStatusCode)
 }
 
 func removeTenantExpectStatusCode(t *testing.T, providedTenant Tenant, expectedStatusCode int) {
-	makeTenantRequestExpectStatusCode(t, providedTenant, http.MethodDelete, expectedStatusCode)
+	makeTenantRequestExpectStatusCode(t, providedTenant, http.MethodDelete, config.TenantFetcherFullURL, expectedStatusCode)
 }
 
-func makeTenantRequestExpectStatusCode(t *testing.T, providedTenant Tenant, httpMethod string, expectedStatusCode int) {
-	byteTenant, err := json.Marshal(providedTenant)
-	require.NoError(t, err)
+func removeRegionalTenantExpectStatusCode(t *testing.T, providedTenant Tenant, expectedStatusCode int) {
+	makeTenantRequestExpectStatusCode(t, providedTenant, http.MethodDelete, config.TenantFetcherFullRegionalURL, expectedStatusCode)
+}
 
-	request, err := http.NewRequest(httpMethod, config.TenantFetcherFullURL, bytes.NewBuffer(byteTenant))
-	require.NoError(t, err)
-	request.Header.Add("Authorization", fmt.Sprintf("Bearer %s", fetchToken(t)))
+func makeTenantRequestExpectStatusCode(t *testing.T, providedTenant Tenant, httpMethod, url string, expectedStatusCode int) {
+	request := createTenantRequest(t, providedTenant, httpMethod, url)
 
 	response, err := httpClient.Do(request)
 	require.NoError(t, err)
 	require.Equal(t, expectedStatusCode, response.StatusCode)
 }
 
+func createTenantRequest(t *testing.T, tenant Tenant, httpMethod string, url string) *http.Request {
+	var (
+		body = "{}"
+		err  error
+	)
+
+	if len(tenant.TenantID) > 0 {
+		body, err = sjson.Set(body, config.TenantIDProperty, tenant.TenantID)
+		require.NoError(t, err)
+	}
+	if len(tenant.SubaccountID) > 0 {
+		body, err = sjson.Set(body, config.SubaccountTenantIDProperty, tenant.SubaccountID)
+		require.NoError(t, err)
+	}
+	if len(tenant.CustomerID) > 0 {
+		body, err = sjson.Set(body, config.CustomerIDProperty, tenant.CustomerID)
+		require.NoError(t, err)
+	}
+	if len(tenant.Subdomain) > 0 {
+		body, err = sjson.Set(body, config.SubdomainProperty, tenant.Subdomain)
+		require.NoError(t, err)
+	}
```

Comment:
why not just json marshall the tenant?

Reply:
That was actually the behaviour before, and while testing locally with the props from prod, the tests started failing because they sent for example "subdomain", but our prod configuration requires "subscribedSubdomain" in the tenant-fetcher deployment.